Task_1/Parts/Class.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Task_1
+{
+    class Class
+    {
+        Pupil[] pupils = new Pupil[4];
+
+        //почему нельзя вставить new GoodPupil() в значения по умолчанию?
+        public Class(Pupil first, Pupil second, Pupil third= null, Pupil fourth = null)
+        {
+            this.pupils[0] = first;
+            this.pupils[1] = second;
+            if (third == null)
+                this.pupils[2] = GeneratePuple();
+            else
+                this.pupils[2] = third;
+            if (fourth == null)
+                this.pupils[3] = GeneratePuple();
+            else
+                this.pupils[3] = fourth;
+        }
+
+        /*
+        public Class(Pupil one, Pupil two, Pupil three)
+            :this(one, two, three, new GoodPupil()) {}
+
+        public Class(Pupil one, Pupil two)
+            : this(one, two, new GoodPupil(), new GoodPupil()) { }
+        */
+
```

Comment:
Don't leave any code commented unless you have a reason to do it. Redundant commented code should be deleted